cmake/Hunter/config.cmake

```diff
@@ -0,0 +1,2 @@
+
+hunter_config(opentracing-cpp VERSION ${HUNTER_opentracing-cpp_VERSION} CMAKE_ARGS BUILD_SHARED_LIBS=ON)
```

Comment:
Again, don't believe this is ideal. Hunter already has `HUNTER_BUILD_SHARED_LIBS` to control this. Please remove.

Reply:
Hello @isaachier, HUNTER_BUILD_SHARED_LIBS impacts all the libraries imported by Hunter whereas the above line configure only OpenTracing as a shared Library. If I am not mistaken, this is what is suggested here https://github.com/ruslo/hunter/wiki/example.custom.config.id.

Why do you think is a bad idea?

@ruslo, what do you suggest? When I use HUNTER_BUILD_SHARED_LIBS, all libraries are considered as shared. The problem is that yaml-cpp does not build in such case.

Reply:
> When I use HUNTER_BUILD_SHARED_LIBS, all libraries are considered as shared. The problem is that yaml-cpp does not build in such case

Homogeneous approach (all static or all shared) is correct and easy to check, so `HUNTER_BUILD_SHARED_LIBS` should be used in most cases. If there are some troubles with `yaml-cpp` package then this package should be fixed.

Reply:
@ruslo I have one case where that is not ideal. This package is used as a plugin for https://github.com/envoyproxy/envoy. To do that, I need one big shared object with all dependencies packaged into it. I'll admit that it is pretty rare to use this pattern.

Reply:
One problem with YAML is that under MSVC, there is a problem with a macro definition. I issued a PR for that https://github.com/hunter-packages/yaml-cpp/pull/4

Reply:
Great. I am happy to see this was merged. We will need to prepare a new release of Hunter to land this review, but let's finish this part first.

Reply:
I did a new fix in Yaml-cpp this week. Hunter updated https://github.com/ruslo/hunter/releases/tag/v0.23.202